lib/services/deviceService.js

```diff
@@ -153,11 +157,85 @@ function sendRegistrations(unregister, deviceData, callback) {
 function processContextRegistration(deviceData, body, callback) {
     var newDevice = _.clone(deviceData);
 
-    newDevice.registrationId = body.registrationId;
+    if (body) {
+        newDevice.registrationId = body.registrationId;
+    }
 
     callback(null, newDevice);
 }
 
+
+function createInitialEntityHandler(deviceData, newDevice, callback) {
```

Comment:
First things first, all the linter checks have been passed, do you see any errors? Concerning JSDocs, functions DO include documentation (check the rest of the code) :D except for this case that I'll gladly fix.